### PR TITLE
4.9 Changelog

### PR DESCRIPTION
**Pull request by amyblais**

### CODE REVIEW

File: source/administration/changelog.md [hunk 1]
```diff
@@ -4,6 +4,154 @@ This changelog summarizes updates to [Mattermost Team Edition](http://www.matter
...
+ - Migrated system permissions to roles, which allows for more granular permissions at team and channel level, as well as custom roles, in future Mattermost releases.
+ - Several configuration settings were migrated to roles in the database where changing their `config.json` values no longer take effect. See the compatibility section of this changelog to learn more.
+ 
+#### Global Relay (Beta) ([Enterprise Edition E20](https://about.mattermost.com/pricing/) Add-On)
+ - Added compliance export support for Global Relay as a compliance solution. [Learn more here](https://about.mattermost.com/default-compliance-export-documentation).
+
+### Improvements
+
+#### Web User Interface
+ - Users can now set their timezone in **Account Settings > Timezone**.
```
Comment (lindy65) [inline source/administration/changelog.md]: "timezone" should be two words but I guess we should leave it here if "Timezone" is shown in "Account Settings"?

File: source/administration/changelog.md [hunk 2]
```diff
@@ -4,6 +4,152 @@ This changelog summarizes updates to [Mattermost Team Edition](http://www.matter
...
+ 
+#### Team Icons
+ - Added support for team icons in the team sidebar, within the border of the existing team icons.
+
+#### Advanced Permissions ([Enterprise Edition E20](https://about.mattermost.com/pricing/))
+ - Migrated system permissions to roles, which allows for more granular permissions at team and channel level, as well as custom roles, in future Mattermost releases.
+ - Several configuration settings were migrated to roles in the database where changing their `config.json` values no longer take effect. See the compatibility section of this changelog to learn more.
+ 
+#### Global Relay (Beta) ([Enterprise Edition E20](https://about.mattermost.com/pricing/) Add-On)
+ - Added compliance export support for Global Relay as a compliance solution. [Learn more here](https://about.mattermost.com/default-compliance-export-documentation).
```
Comment (jwilander) [inline source/administration/changelog.md]: 0/5 but we say "compliance" twice in this sentence, sounds a bit weird

File: source/administration/changelog.md [hunk 3]
```diff
@@ -4,6 +4,152 @@ This changelog summarizes updates to [Mattermost Team Edition](http://www.matter
...
+### Improvements
+
+#### Web User Interface
+ - Users can now set their timezone in **Account Settings > Timezone**.
+ - Added ability for users to load more recent messages after jumping to a search result.
+ - Cursor now returns to the reply thread input box after deleting a reply on the right-hand sidebar.
+
+#### Performance
+ - Decreased channel load time by optimizing the `getParentsPosts` database query.
+ - Decreased load time of large channels with 5,000+ messages by up to 90% by optimizing the `formatPostInChannel` and `postsInThread` database queries.
```
Comment (jwilander) [inline source/administration/changelog.md]: `formatPostInChannel` and `postsInThread` are not database queries, they are client functions

Comment (jwilander) [reply source/administration/changelog.md]: Probably better to say:
```
... by optimizing many client functions related to rendering posts and threads.
```

File: source/administration/changelog.md [hunk 4]
```diff
@@ -4,6 +4,152 @@ This changelog summarizes updates to [Mattermost Team Edition](http://www.matter
...
+
+### Improvements
+
+#### Web User Interface
+ - Users can now set their timezone in **Account Settings > Timezone**.
+ - Added ability for users to load more recent messages after jumping to a search result.
+ - Cursor now returns to the reply thread input box after deleting a reply on the right-hand sidebar.
+
+#### Performance
+ - Decreased channel load time by optimizing the `getParentsPosts` database query.
```
Comment (jwilander) [inline source/administration/changelog.md]: Nobody knows what the `getParentsPosts` query is, better to say something like:

```
... by optimizing database queries used to fetch threads and parent posts in channel.
``

File: source/administration/changelog.md [hunk 5]
```diff
@@ -4,6 +4,152 @@ This changelog summarizes updates to [Mattermost Team Edition](http://www.matter
...
+ - Fixed plugin webapp manifest field 'bundle_path' to match documentation and be the path to the plugin's JavaScript bundle relative to the root of the plugin, similar to the server manifest 'executable' field.
+ - Fixed an issue where Direct Message channel with yourself did not show up in channel switcher.
+ - Fixed an issue where deleting one username from "add member to a channel" field deleted all names.
+ - Fixed an issue where link was missing in an ephemeral message to add mentioned user to the channel.
+ - Fixed an issue where View/Manage members should have been sorted by username, not online status.
+ - Fixed an issue where a non-system-admin should not see `Is Trusted` option on OAuth 2.0 integrations.
+ - Fixed an issue with being unable to click on pinned post, channel members, and so on with keyboard focus on search box.
+ - Fixed an issue where Mattermost only imported first user during Slack import.
+ - Fixed an issue where cleared search term reappeared after closing RHS.
+ - Fixed an issue where image was posted while RHS was open with an oversized thumbnail.
```
Comment (esethna) [inline source/administration/changelog.md]: unclear wording

File: source/administration/changelog.md [hunk 6]
```diff
@@ -4,6 +4,152 @@ This changelog summarizes updates to [Mattermost Team Edition](http://www.matter
...
+ - Fixed an issue where both "test" and "test.example" were added to channel when adding "test.example" user to a channel via ephemeral message that appears when mentioning someone not in a channel.
```
Comment (esethna) [inline source/administration/changelog.md]: Example is too specific. Propose:

Fixed an issue with adding users to channels when the usernames contained periods.

File: source/administration/changelog.md [hunk 7]
```diff
@@ -4,6 +4,152 @@ This changelog summarizes updates to [Mattermost Team Edition](http://www.matter
...
+ - Fixed an issue with a JavaScript error when using CMD/CTRL-K keyboard shortcut to change channels.
```
Comment (esethna) [inline source/administration/changelog.md]: Did this exist in 4.8? If not we shouldn't have it in this list

File: source/administration/changelog.md [hunk 8]
```diff
@@ -4,6 +4,152 @@ This changelog summarizes updates to [Mattermost Team Edition](http://www.matter
...
+ - Fixed an issue with not being able to get past second page of `/admin_console/users`.
+ - Fixed an issue where ALT+UP/DOWN caused error in console and then stopped working.
```
Comment (esethna) [inline source/administration/changelog.md]: Did this exist in 4.8? If not we shouldn't have it in this list

File: source/administration/changelog.md [hunk 9]
```diff
@@ -4,6 +4,152 @@ This changelog summarizes updates to [Mattermost Team Edition](http://www.matter
...
+     - Added `"ScopingIDPName": ""`, to add the name associated with a user's Scoping Identity Provider ID.
+ - Under `DisplaySettings"` in `config.json`:
+     - Added `"ExperimentalTimezone": false`, to allow selecting the timezone used for timestamps in the user interface and email notifications.
+
+#### API Changes
+
+ - It is required that any new integrations use API v4 endpoints. For more details, and for a complete list of available endpoints, see [https://api.mattermost.com/](https://api.mattermost.com/).
+ - All API v3 endpoints have been deprecated and are scheduled for removal in Mattermost v5.0.
+ 
+##### RESTful API v4 Changes
```
Comment (esethna) [inline source/administration/changelog.md]: Can these headings be removed? Or is this section still WIP?

File: source/administration/changelog.md [hunk 10]
```diff
@@ -4,6 +4,152 @@ This changelog summarizes updates to [Mattermost Team Edition](http://www.matter
...
+ - Several configuration settings were migrated to roles in the database where changing their `config.json` values no longer take effect. See the compatibility section of this changelog to learn more.
+ 
+#### Global Relay (Beta) ([Enterprise Edition E20](https://about.mattermost.com/pricing/) Add-On)
+ - Added export support for Global Relay as a compliance solution. [Learn more here](https://about.mattermost.com/default-compliance-export-documentation).
+
+### Improvements
+
+#### Web User Interface
+ - Users can now set their timezone in **Account Settings > Timezone**.
+ - Added ability for users to load more recent messages after jumping to a search result.
```
[the remark this answers is not in the document]
Comment (amyblais) [reply source/administration/changelog.md]: I can't find the PR for some reason. Also there is a Jira ticket related to this still open so not sure what happened here. Removed it from here for now.

File: source/administration/changelog.md [hunk 11]
```diff
@@ -4,6 +4,152 @@ This changelog summarizes updates to [Mattermost Team Edition](http://www.matter
...
+ - Added a "Test Connection" button to test Amazon S3 connection.
+ - Added a `platform user email` command to change a user's email address.
+
+#### Enterprise Edition
+ - When `ExperimentalTownSquareIsReadOnly` is set to `true`, non-admins can no longer react to messages, pin messages or update channel information.
+ - Added cache invalidation totals to [Performance Monitoring](https://docs.mattermost.com/deployment/metrics.html).
+
+### Bug Fixes
+
+ - Fixed 404 errors about custom emoji in the logs.
```
[the remark this answers is not in the document]
Comment (amyblais) [reply source/administration/changelog.md]: Reworded it - however, the Jira tickets don't give much detail: https://mattermost.atlassian.net/browse/MM-10010 and https://mattermost.atlassian.net/browse/MM-9653.

File: source/administration/changelog.md [hunk 12]
```diff
@@ -4,6 +4,152 @@ This changelog summarizes updates to [Mattermost Team Edition](http://www.matter
...
+#### Advanced Permissions ([Enterprise Edition E20](https://about.mattermost.com/pricing/))
+ - Migrated system permissions to roles, which allows for more granular permissions at team and channel level, as well as custom roles, in future Mattermost releases.
+ - Several configuration settings were migrated to roles in the database where changing their `config.json` values no longer take effect. See the compatibility section of this changelog to learn more.
+ 
+#### Global Relay (Beta) ([Enterprise Edition E20](https://about.mattermost.com/pricing/) Add-On)
+ - Added export support for Global Relay as a compliance solution. [Learn more here](https://about.mattermost.com/default-compliance-export-documentation).
+
+### Improvements
+
+#### Web User Interface
```
[the remark this answers is not in the document]
Comment (amyblais) [reply source/administration/changelog.md]: I can't fin the PR, would you know which PR it is? I'll continue looking for it too though.

File: source/administration/changelog.md (hunk 9 above)
Comment (amyblais) [reply source/administration/changelog.md]: WIP, waiting for Derrick.

Comment (amyblais) [reply source/administration/changelog.md]: Done now.

**Pull request merged**